plugins/inputs/modbus/modbus.go

```diff
@@ -562,6 +580,7 @@ func (m *Modbus) Gather(acc telegraf.Accumulator) error {
 
 	err := m.getFields()
 	if err != nil {
+		disconnect(m)
```

Comment:
Why do we need to disconnect?

Reply:
Because if something wrong happened we need to close the connection, 

for example, if I turn off my serial controller and then turn on, I get multiple errors about the connection.